kotest-property/kotest-property-core/src/commonMain/kotlin/io/kotest/property/core/runtime.kt

```diff
@@ -0,0 +1,131 @@
+package io.kotest.property.core
+
+import io.kotest.assertions.print.print
+import io.kotest.common.ExperimentalKotest
+import io.kotest.core.names.TestName
+import io.kotest.core.spec.style.scopes.RootScope
+import io.kotest.core.spec.style.scopes.addTest
+import io.kotest.property.AssumptionFailedException
+import io.kotest.property.PropertyTesting
+import io.kotest.property.ShrinkingMode
+import io.kotest.property.internal.Counter
+import io.kotest.property.internal.ShrinkResult
+import io.kotest.property.internal.throwPropertyTestAssertionError
+
+@ExperimentalKotest
+fun RootScope.proptest(name: String, configure: PropTestContext.() -> Unit) {
```

Comment:
I have a suggestion of something to look at in the future, when the POC is more refined, is to add a `@DslMarker`, to try and prevent a property test being called within a property test.

https://kotlinlang.org/docs/type-safe-builders.html#scope-control-dslmarker

Reply:
Yeah I planned on that (plus stopping the gen delegate being defined inside the property block)